relay-dynamic-config/src/metrics.rs

```diff
@@ -124,3 +125,178 @@ impl TransactionMetricsConfig {
         self.version > 0 && self.version <= TRANSACTION_EXTRACT_VERSION
     }
 }
+
+/// Configuration for generic extraction of metrics from all data categories.
+#[derive(Clone, Default, Debug, Serialize, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct MetricExtractionConfig {
+    /// Versioning of metrics extraction. Relay skips extraction if the version is not supported.
+    pub version: u16,
+
+    /// A list of metric specifications to extract.
+    #[serde(default)]
+    pub metrics: Vec<MetricSpec>,
+
+    /// A list of tags to add to previously extracted metrics.
+    ///
+    /// These tags add further tags to a range of metrics. If some metrics already have a matching
+    /// tag extracted, the existing tag is left unchanged.
+    #[serde(default)]
+    pub tags: Vec<TagMapping>,
+}
+
+impl MetricExtractionConfig {
+    /// Returns `true` if metric extraction is disabled.
+    pub fn is_empty(&self) -> bool {
+        self.version == 0 || (self.metrics.is_empty() && self.tags.is_empty())
+    }
+}
+
+/// Specification for a metric to extract from some data.
+#[derive(Clone, Debug, Serialize, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct MetricSpec {
+    /// Category of data to extract this metric for.
+    pub category: DataCategory,
+
+    /// Identifier of the metric to extract.
+    pub mri: String,
+
+    /// A path to the field to extract the metric from.
+    ///
+    /// This value contains a fully qualified expression pointing at the data field in the payload
+    /// to extract the metric from. It follows the
+    /// [`FieldValueProvider`](relay_sampling::FieldValueProvider) syntax that is also used for
+    /// dynamic sampling.
+    ///
+    /// How the value is treated depends on the metric type:
+    ///
+    /// - **Counter** metrics are a special case, since the default product counters do not count
+    ///   any specific field but rather the occurrence of the event. As such, there is no value
+    ///   expression, and the field is set to `None`. Semantics of specifying remain undefined at
+    ///   this point.
+    /// - **Distribution** metrics require a numeric value. If the value at the specified path is
+    ///   not numeric, metric extraction will be skipped.
+    /// - **Set** metrics are require a string value, which is then emitted into the set as unique
+    ///   value. Insertion of numbers and other types is undefined.
+    ///
+    /// If the field does not exist, extraction is skipped.
+    #[serde(default)]
+    pub field: Option<String>,
+
+    /// An optional condition to apply before extraction.
```

Comment:
nit:
```suggestion
    /// An optional condition to meet before extraction.
```
I think `apply` may lead to some ambiguity, while `meet` doesn't.